### PR TITLE
Rewrite 10 holograph tests

### PR DESCRIPTION
**Pull request by mmvazzano**
Add `/tests/foudnry/deploy/10_holograph_tests.t.sol`, which is the Foundry transcription for `/test/10_holograph_tests.t.sol`.
Comments in NatSpec format are included.

### CODE REVIEW

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 1]
```diff
@@ -0,0 +1,700 @@
...
+    * @dev The tests include verifying initialization, admin functions, revert behaviors, and external contract interactions.
+    * The tests check the validity of various contract slots like bridge, chainId, factory, interfaces, operator, registry, 
+    * treasury, and utilityToken.
+    * The tests also verify that only the admin can alter certain contract slots, while non-owners trigger revert behaviors.
+    * Translation of a suite of Hardhat tests found in test/10_holograph_tests.ts
+    */
+
+contract HolographTests is Test {
+    address admin = vm.addr(1);
+    address user = vm.addr(2);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol:22]: add new var address  origin  with value '0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38'

File: test/foundry/deploy/10_holograph_tests.t.sol [hunk 2]
```diff
@@ -0,0 +1,700 @@
...
+    * @notice Test the setAdmin function of the Holograph contract.
+    * @dev This test is designed to verify the functionality of setting a new admin address in the contract.
+    * This test retrieves the current admin address from the Holograph contract,
+    * performs a prank operation on the admin address using the VM,
+    * and then sets a new admin address to the contract.
+    */
+    // TODO Admin does not correspond to the address that performs the init
+    function testSetAdmin() public {
+        admin = holograph.getAdmin();
+        vm.prank(admin);
```
Comment (fgregoribattista) [inline test/foundry/deploy/10_holograph_tests.t.sol]: use the new var origin in all these cases where it is necessary to change the contract call to the owner.

**Comment by fgregoribattista** [comment]
very good work, try to change the corrections and for me its ready to review and merge

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 1 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol:22]: Done

File: test/foundry/deploy/10_holograph_tests.t.sol (hunk 2 above)
Comment (mmvazzano) [reply test/foundry/deploy/10_holograph_tests.t.sol]: Done

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**